controllers/progressivesync_controller.go

```diff
@@ -99,11 +107,25 @@ func (r *ProgressiveSyncReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	pss, _ := r.StateManager.Get(ps.Name)
+
+	newHash, err := r.calculateHashedSpec(ctx, &ps)
+	if err != nil {
+		log.Error(err, "Failed to generate the hash value from the ApplicationSet spec")
```

Comment:
Shouldn't we return here?

Reply:
I was thinking that we can add the logic to return/fail once we start working on the issue for the progress deadline. But I can add it now as well.

Reply:
I have now updated it to return the error we got above. I will resolve for now 🙂